### PR TITLE
Theoretical biology revision prep

### PR DESCRIPTION
**Pull request by drvinceknight**

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -60,23 +60,13 @@
...
-memory-one strategy can do. These strategies emerged from an evolutionary
-selection process and their existence contradicts widely-repeated folklore in
-the evolutionary game theory literature: complex strategies can be
-extraordinarily effective, zero-determinant strategies can be outperformed by
-non-zero determinant strategies, and longer memory strategies are able to
-outperform short memory strategies. Moreover, while resistance to extortion is
-critical for the evolution of cooperation, the extortion of weak opponents
-need not prevent cooperation between stronger opponents, and this adaptability
-may be crucial to maintaining cooperation in the long run.
+The main contribution of this paper is that extortionate strategies as defined
```
Comment (marcharper) [inline main.tex]: Is that right? I think that was already known, the main contribution is being able to expand the definition of extortionate to arbitrary strategies, and then to show that this phenomenon still holds.

File: main.tex [hunk 2]
```diff
@@ -102,19 +93,24 @@
...
-ichinose2018zero, Moran1707} the true effectiveness of these strategies in an
-evolutionary setting was discussed. For example~\cite{adami2013evolutionary}
-showed that ZD strategies were not evolutionarily stable. Furthermore, in that
-work it was also postulated that `evolutionarily successful ZD strategies could
-be designed that use longer memory to distinguish self from non-self'. In a non
+In~\cite{adami2013evolutionary, hilbe2018partners, Hilbe2013, hilbe2013adaptive,
+hilbe2015partners, ichinose2018zero, Moran1707} the true effectiveness of these
+strategies in an evolutionary setting was discussed. For
+example~\cite{adami2013evolutionary} showed that ZD strategies were not
+evolutionarily stable. Furthermore, in that work it was also postulated that
```
Comment (marcharper) [inline main.tex:111]: One of our ML papers also showed that evolutionary successful strategies can learn to self recognize. My "Art of War" paper shows that one can design strategies to do so, albeit not necessarily ZD strategies (no even memory one).

File: main.tex [hunk 3]
```diff
@@ -102,19 +93,24 @@
...
+`evolutionarily successful ZD strategies could be designed that use longer
+memory to distinguish self from non-self'.  In~\cite{hilbe2018partners} two sets
+of strategies are identified: partners and rivals and some discussion about the
+environments necessary for either to be evolutionary stable are given.
+
+In a non
```
Comment (marcharper) [inline main.tex]: need hyphen "non-evolutionary"

File: main.tex [hunk 4]
```diff
@@ -173,33 +169,42 @@
 
 \section{Methods: Recognising Extortion}\label{sec:sserror-zd-strategies}
 
-Zero-determinant strategies are a special case of memory-one strategies,
+The role of this section is to review the definition of ZD
```
Comment (marcharper) [inline main.tex]: This section reviews ...., presents, ...

In general I'm not a fan of a section self-summarizing at the beginning, but perhaps this is responding to a specific comment

File: main.tex [hunk 5]
```diff
@@ -393,24 +399,29 @@ \subsection{Measuring Extortion from the History of Play}
...
 is even feasible (as noted in~\cite{Press2012}), a
 high \(\SSe\) does imply that a strategy is not extortionate. For a measured
-\(p\), \(\SSe\) corresponds to the best fitting \(\alpha, \beta\). Suspicion of
+\(\textbf{p}\), \(\SSe\) corresponds to the best fitting \(\alpha, \beta\). Suspicion of
 extortion then corresponds to a threshold on \(\SSe\) and a comparison of the
 measured \(\chi=\frac{-\beta}{\alpha}\).
 
 \section{Results: Validation of approach and Numerical experiments}\label{sec:numerical-experiments}
 
+The role of this section is to validate the approach of the previous section and
```
Comment (marcharper) [inline main.tex]: This section validates ...., and presents ...

File: main.tex (hunk 1 above)
Comment (marcharper) [reply main.tex]: You say something similar around round line 575

File: main.tex [hunk 6]
```diff
@@ -729,6 +765,25 @@ \section{Discussion}\label{sec:conclusion}
 in complex strategies themselves, which is demonstrable not the case. In fact,
 complex strategies can be the most effective against a diverse set of opponents.
 
+There are various potential applications for \(\SSe\), one of which could be to
+devise an agent that learns during the interactions with another agent.
+Figure~\ref{fig:learning_sse} shows the average \(\SSe\) value over a number of
+iterations over a number of repetitions. More investigation would be required
```
Comment (marcharper) [inline main.tex:776]: Maybe rephrase what "would be interesting" to do to anything concrete we can say now. Otherwise the reviewers may ask that we do more of these ...

File: reviews/2022-04-25.md [hunk 7]
```diff
@@ -1,16 +1,33 @@
...
+improvement of the presentation of the work: we aimed to demonstrate in Section
+2 that there was a body of literature that claimed that extortion is highly
+effective. Section 3 is our main contribution to the literature which is to
+highlight that such claims are incomplete and in fact that extortion lacks the
+ability to adapt which is fundamentally important to evolution.
+
+We believe we have improved the presentation of this and will discuss the
+modifications made in detail below.
+
+Reviewer 2s made 2 main suggestions for further work, we have added to the paper
```
Comment (marcharper) [inline reviews/2022-04-25.md]: 2s --> 2

File: reviews/2022-04-25.md [hunk 8]
```diff
@@ -97,9 +180,19 @@ clear presentation, interesing (although limited) idea and concrete simulation.
...
+   of work.
+
 2. Based on zd strategies, more works about control the other's payoff have been
    seen. For example, [Akin 2015, "What you gotta know to play good in the
    iterated prisoners dilemma"] designed the non-linear control and this is
    further formulated by Hao et al. 2018 "Payoff Control in Iterated Prisoners'
    Dilemma". Is it possible for this paper to handle the non-linear control
    cases that proposed in these papers？
+
+   TODO: Read to see if we can do this.
```
Comment (marcharper) [inline reviews/2022-04-25.md]: This is also out of scope IMO

**Comment by marcharper** [comment]
Additional feedback:

* The plots in Figure 2 are too small, let's try a 2x2 instead of a 1x4 layout 
* Figure 3: do we want to add a best fit curve of some kind?
* Figure 8: Maybe expand the caption and call out ZD-SET-2 as generous

**Review by marcharper — CHANGES_REQUESTED** [review]

File: reviews/2022-04-25.md [hunk 9]
```diff
@@ -1,16 +1,33 @@
...
+of the work and not the scientific substance which it seems that both reviewers
+support. Indeed, the major point being
+about the contradiction between Sections 2 and 3 identify a possibility for
+improvement of the presentation of the work: we aimed to demonstrate in Section
+2 that there was a body of literature that claimed that extortion is highly
+effective. Section 3 is our main contribution to the literature which is to
+highlight that such claims are incomplete and in fact that extortion lacks the
+ability to adapt which is fundamentally important to evolution.
+
+We believe we have improved the presentation of this and will discuss the
```
Comment (Nikoleta-v3) [inline reviews/2022-04-25.md]: It's not clear what `this` refers to. Merge this sentence with the previous paragraph?

File: reviews/2022-04-25.md [hunk 10]
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
...
+   > positive skew of SSE is not favored by evolution. In the first part of the
+   > manuscript (Sec. 2), the authors mainly focused on the effectiveness of
+   > extortion (how it is defined from the history of gameplay and how it is
+   > strong). But in Sec. 3, the authors revealed the ineffectiveness of
+   > extortion. Thus, I felt the contradiction between the former and the latter
+   > parts in the manuscript.
+
+   The role of Section 2 is to give an overview of the mathematical definition
+   of Extortionate strategies so as to present the novel linear algebraic
+   measure of the our manuscript. We do not believe that we mainly focus on the
```
Comment (Nikoleta-v3) [inline reviews/2022-04-25.md]: ~the~

File: reviews/2022-04-25.md [hunk 11]
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
...
+   > 469-477 (2018)]. On the other hand, generous ZD strategies [Stewart &
+   > Plotkin, PNAS, 110, 15348-15353 (2013)], which are very weak in one to one
+   > interaction but are very strong in an evolving population, have attracted
+   > great attention in the literature. Thus, the analysis of generous ZD
+   > strategies should be included in this manuscript. How is the SSE of generous
+   > ZD and is that a negative skew or positive skew?
+
+   We have added the Hilbe et al. 2018 paper to our introduction in the
+   paragraph that had other papers that questioned the effectiveness of ZD
+   strategies in evolving populations. Thank you for this suggestion.
```
Comment (Nikoleta-v3) [inline reviews/2022-04-25.md]:
```suggestion
   We have added the Hilbe et al. 2018 paper to our introduction where we
   discuss  other papers that have questioned the effectiveness of ZD
   strategies in evolving populations. Thank you for this suggestion.
```

File: reviews/2022-04-25.md [hunk 12]
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
...
+
+   The 2013 work of Stewart and Plotkin is in the paper and in is the subject of
```
Comment (Nikoleta-v3) [inline reviews/2022-04-25.md]:
```suggestion
   The 2013 work of Stewart and Plotkin is in the paper and it is the subject of
```

File: main.tex [hunk 13]
```diff
@@ -93,6 +83,7 @@
     \label{eqn:theipd}
```
Comment (Nikoleta-v3) [inline main.tex:78]: regarding the comment:

> The payoff matrix
> of PD in the Introduction and the replicator dynamics described as Eq. (25)
> are boorish because each variable is not explained anywhere.

Should we include more discussion regarding the payoff matrix? That `R` stand for the `Reward` payoff etc?

File: main.tex [hunk 14]
```diff
@@ -173,33 +169,42 @@
 
 \section{Methods: Recognising Extortion}\label{sec:sserror-zd-strategies}
 
-Zero-determinant strategies are a special case of memory-one strategies,
+The role of this section is to review the definition of ZD
+strategies from the literature, present the vector space in which such
+strategies exist and finally present a novel measure that allows for a measure
+of how far any memory one strategy is from the space of ZD
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion
of how far any memory-one strategy is from the space of ZD
```

File: main.tex [hunk 15]
```diff
@@ -547,23 +563,37 @@ \subsection{Numerical experiments}
...
+
+In the original work introducing ZD strategies~\cite{Press2012}, effectiveness
+in evolutionary settings was already considered. Since then, most work
+surrounding these strategies considers their performance in evolutionary
+settings. Examples include~\cite{adami2013evolutionary, hilbe2018partners,
+Hilbe2013, hilbe2013adaptive, hilbe2015partners, ichinose2018zero, Moran1707}.
+The main motivation for this consideration is to gain insights on to how
+behaviours might arise but also whether or not they are stable in various
+settings such as social and biological interactions. Most of such work
+considers the space of memory one strategies alone, in this paper a wider
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion
considers the space of memory-one strategies alone, in this paper a wider
```

File: main.tex (hunk 1 above)
Comment (drvinceknight) [reply main.tex]: Yes, you're right 👍

File: main.tex (hunk 2 above)
Comment (drvinceknight) [reply main.tex:111]: Will add these. Thank you.

File: main.tex (hunk 4 above)
Comment (drvinceknight) [reply main.tex]: I'm not a fan at all either, this was specifically requested by reviewer 1.

File: reviews/2022-04-25.md (hunk 9 above)
Comment (drvinceknight) [reply reviews/2022-04-25.md]: Yup, thanks.

File: main.tex (hunk 13 above)
Comment (drvinceknight) [reply main.tex:78]: Yup! I think I meant to do this but forgot. Thanks.

File: reviews/2022-04-25.md (hunk 10 above)
Comment (drvinceknight) [reply reviews/2022-04-25.md]: Actually I disagree, have left `... focus on the effectiveness` here. Let me know if I'm missing something.

**Comment by drvinceknight** [comment]
> The plots in Figure 2 are too small, let's try a 2x2 instead of a 1x4 layout

Done, I think it looks good, let me know what you think.

> Figure 3: do we want to add a best fit curve of some kind?

I've added a line. Played around with various other polynomials but I think the trend is the main thing that looked interesting. Happy to try something else though.

> Figure 8: Maybe expand the caption and call out ZD-SET-2 as generous

Done :)


I'll spend some time tomorrow tidying everything up but any other comments or things you'd like me to try please let me know :) 👍

File: reviews/2022-04-25.md (hunk 10 above)
Comment (Nikoleta-v3) [reply reviews/2022-04-25.md]: Sorry my comment wasn't very clear, it took me a while to see it too. `measure of the our manuscript.` either remove `the` or `our`

Comment (drvinceknight) [reply reviews/2022-04-25.md]: Thank you!

**Comment by drvinceknight** [comment]
I think I have modified everything and this is essentially good to go. I have added a marked-up pdf: https://github.com/drvinceknight/testing_for_ZD/blob/c00f8800a16b4c293b9d5dbdadbe9eeb2e3e65c7/diff.pdf

Let me know if you have any other thoughts (including if you think I could do a better job at anything you've already suggested).

**Comment by marcharper** [comment]
The plots look much better. I opened a PR with some copy edits.

**Comment by drvinceknight** [comment]
> I opened a PR with some copy edits.

Thank you! I have pushed a new version of the diff'd pdf.

**Comment by drvinceknight** [comment]
I'm going to go ahead and merge this. Will plan to resubmit on Monday. Thanks both 👍

**Comment by Nikoleta-v3** [comment]
Thank you @drvinceknight ! 🎉
